api/chunks/utils.py

```diff
@@ -0,0 +1,30 @@
+from documents.models import Document
+
+
+def split_document_into_chunks(document: Document, chunk_size: int = 100) -> list[dict]:
+    """Splits document into chunks of size chunk_size and returns them as dictionaries.
+    End_char is assumed to be the last included character of the document."""
+    chunks = []
+    start_char = 0
+    end_char = chunk_size
+    chunk_idx = 1
+
+    while start_char < len(document.text):
+        if end_char >= len(document.text):
+            end_char = len(document.text)
+        chunk_text = document.text[start_char:end_char]
+        chunk = dict(
+            text=chunk_text,
+            chunk_idx=chunk_idx,
+            start_char=start_char,
+            end_char=(end_char - 1),
+            document_idx=document.id,
+        )
+
+        chunks.append(chunk)
+
+        start_char += chunk_size
+        end_char += chunk_size
+        chunk_idx += 1
+
+    return chunks
```

Comment:
**suggestion:** there's a lot of places where you can achieve off-by-one errors this way, we can use more automatic way (this code seems to work fine in the REPL, but make sure it's formatted etc., as I wrote it in the suggestion box xD)

I don't really like solution with `next_start_char`, but I don't have any better idea right now (as we have to check chunk's length anyway for this)
```suggestion
from documents.models import Document
from itertools import batched


def split_document_into_chunks(document: Document, chunk_size: int = 100) -> list[dict]:
    """Splits document into chunks of size chunk_size and returns them as dictionaries.
    End_char is assumed to be the last included character of the document."""
    
    chunks = []
    start_char = 0
    
    for i, chunk in enumerate(batched(document.text, chunk_size)):
        next_start_char = start_char + len(chunk)
        
        chunks.append({
            "text": "".join(chunk),
            "chunk_idx": i,
            "start_char": start_char,
            "end_char": next_start_char - 1,
            "document_idx": document.id,
        })
        
        start_char = next_start_char
    
    return chunks
```